### PR TITLE
Homework2: Aleksandr Strokov

### PR DESCRIPTION
**Pull request by alstrokov**

### CODE REVIEW

File: src/task01.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""
+Determine if given number is polynome.
+"""
+num = int(input("Please, enter number to test\n"))
+
+ditgits = []
+while num:
+    ditgits.append(num % 10)
```
Comment (belovmd) [inline src/task01.py]: please create a reversed number using arithmetic operations

File: src/task01.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    num //= 10
+
+while ditgits:
+    if ditgits[:1] != ditgits[-1:]:
```
Comment (belovmd) [inline src/task01.py]: please use numbers comparison

File: src/task03.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+for i in range(1, 101):
+    if not i % 5 and not i % 3:
```
Comment (belovmd) [inline src/task03.py]: can we use 15 here?

File: src/task04.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+"""
+Subtask2:
+to square every digit of a number.
+For example, if we run 9119 through the function, 811181 will come out,
+because 9**2 is 81 and 1**2 is 1.
+"""
+
+
+def square_digits(num):
+    s = str(num)
```
Comment (belovmd) [inline src/task04.py]: please avoid magic letters. it's not clear for me names s, c, v

File: src/task04.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+continue reducing in this way until a single-digit number is produced.
+This is only applicable to the natural numbers.
+"""
+
+
+def num2collection(num) -> list:
+    rez = []
+
+    while num:
+        rez.append(num % 10)
```
Comment (belovmd) [inline src/task04.py]: please don't use lists here. it would be enough to make a number

**Review by belovmd — CHANGES_REQUESTED** [review]
1. please add tasks description to docstrings
2. Please name PR according our guidelines
3. please address these issues:

**Comment by alstrokov** [comment]
Homework2 updated

File: src/task01.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+"""Determine if given number is polynome."""
+
+
+def isPoly(number):
```
Comment (belovmd) [inline src/task01.py:4]: java naming, not python

File: src/task01.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+"""Determine if given number is polynome."""
```
Comment (belovmd) [inline src/task01.py:1]: palindrome

File: src/task03.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+Buzz if number devided by 5,
+FizzBuzz if number devided by 3 and by 5.
+"""
+
+formatlist = ['', 'Fizz', 'Buzz', 'FizzBuzz']
+
+for num in range(1, 101):
+    formatlist[0] = num
+    print('{}'.format(
+        formatlist[(num % 3 == 0) + 2 * (num % 5 == 0)]))
```
Comment (belovmd) [inline src/task03.py:12]: it looks too complex

File: src/task04.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+"""
+Subtask2:
+to square every digit of a number.
+For example, if we run 9119 through the function, 811181 will come out,
+because 9**2 is 81 and 1**2 is 1.
+"""
+
+
+def square_digits(num):
+    stringRepresentation = str(num)
```
Comment (belovmd) [inline src/task04.py:34]: ''.join(int(el)**2 for el in str(num))

File: src/task04.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+If that value has more than one digit,
+continue reducing in this way until a single-digit number is produced.
+This is only applicable to the natural numbers.
+"""
+
+
+def digital_root(number):
+    while number > 9:
+        digitsSum = 0
+        for digit in str(number):
```
Comment (belovmd) [inline src/task04.py:56]: can't we use arithmetic operations here?

**Review by belovmd — COMMENTED** [review]
7.5

**Pull request closed without merging**